cfme/containers/image.py

```diff
@@ -1,64 +1,52 @@
 # -*- coding: utf-8 -*-
 # added new list_tbl definition
+from navmazing import NavigateToSibling, NavigateToAttribute
+
 from cfme.common import SummaryMixin, Taggable
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import toolbar as tb, CheckboxTable
-from cfme.web_ui.menu import nav
-from . import details_page
+from cfme.web_ui import toolbar as tb, CheckboxTable, paginator, summary_title
+from utils.appliance.endpoints.ui import navigator, CFMENavigateStep
+from utils.appliance import Navigatable
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
 
-nav.add_branch(
-    'containers_images',
-    {
-        'containers_image':
-        lambda ctx: list_tbl.select_row_by_cells(
-            {'Name': ctx['image'].name, 'Provider': ctx['image'].provider.name}),
 
-        'containers_image_detail':
-        lambda ctx: list_tbl.click_row_by_cells(
-            {'Name': ctx['image'].name, 'Provider': ctx['image'].provider.name}),
-    }
-)
+def match_title_and_controller():
+    title_match = sel.execute_script('return document.title;') == 'CFME: Images'
+    controller_match = sel.execute_script('return ManageIQ.controller;') == 'container_image'
+    return title_match and controller_match
 
 
-class Image(Taggable, SummaryMixin):
+class Image(Taggable, SummaryMixin, Navigatable):
 
-    def __init__(self, name, provider):
+    def __init__(self, name, provider, appliance=None):
         self.name = name
         self.provider = provider
+        Navigatable.__init__(self, appliance=appliance)
+
+
+@navigator.register(Image, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
 
-    def _on_detail_page(self):
-        return sel.is_displayed(
-            '//div//h1[contains(., "{} (Summary)")]'.format(self.name))
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Compute', 'Containers', 'Container Images')(None)
 
-    def load_details(self, refresh=False):
-        if not self._on_detail_page():
-            self.navigate(detail=True)
-        elif refresh:
-            tb.refresh()
+    def resetter(self):
+        tb.select('Grid View')
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
 
-    def click_element(self, *ident):
```

Comment:
@psav found that Summary class uses load_details(), which @mfalesni has noted SummaryMixin is preferred over InfoBox. Looks like I'll need to keep that method at least.